src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java

```diff
@@ -175,10 +175,11 @@ private Uni<UpdatedDocument> processUpdate(
                 }
               }
 
-              // otherwise shred
-              JsonNode updatedDocument = documentUpdaterResponse.document();
               final WritableShreddedDocument writableShreddedDocument =
-                  shredder().shred(updatedDocument, readDocument.txnId());
+                  shredder().shred(documentUpdaterResponse.document(), readDocument.txnId());
+
+              // Have to do this because shredder adds _id field to the document if it doesn't exist
+              JsonNode updatedDocument = writableShreddedDocument.docJsonNode();
```

Comment:
Maybe we should improve the shredded document impl and avoid this side effect call?

Reply:
That sounds like a good improvement idea; something that handles insertion and reordering of `_id` before Shredder gets to deal with it.

I'll create a follow-up issue.